2019.10+fio tee

ldts opened the pull request:
Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

ldts committed:
[FIO extras] imx7ulpea: prepare for AVB calls

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

ricardosalveti commented on board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c:
```diff
@@ -4,7 +4,6 @@
  *
  * SPDX-License-Identifier:	GPL-2.0+
  */
-
```
nit: can you not remove this line? Just to avoid possible conflicts later on as we rebase the patches.

ldts replied:
sure, let me repost

ricardosalveti commented on configs/mx7ulpea-ucom_lmp_defconfig:
```diff
@@ -80,3 +80,5 @@ CONFIG_OPTEE_LOAD_ADDR=0x60800000
 CONFIG_OPTEE_TZDRAM_BASE=0x9e000000
 CONFIG_BOOTM_OPTEE=y
+CONFIG_TEE=y
+CONFIG_CMD_SMC=y
```
Can you run 'make mx7ulpea-ucom_lmp_defconfig && make savedefconfig && cp defconfig configs/mx7ulpea-ucom_lmp_defconfig'?

That way the configs will be added at the right order by u-boot itself.

ldts replied:
ok

ldts committed:
[FIO extras] avb: remove fastboot dependency

Work around build problems on imx7ulpea

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

ldts committed:
[FIO extras] avb: avoid imx name collision

ldts committed:
[FIO wip] imx7ulpea: configs: enable AVB (TEE/SMC/RPMB/DCACHE_OFF)

AVB requires TEE and RPMB to implement the required security features.
Having SMC also helps validating TEE from u-boot

Also, somehow, we are having issues invalidting the DCACHE so lets
disable it for integration

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

ldts commented:
added the missing RPMB support required for AVB persistent storage.

mike-scott commented on configs/mx7ulpea-ucom_lmp_defconfig:
```diff
@@ -8,6 +8,7 @@ CONFIG_NR_DRAM_BANKS=1
 CONFIG_SPL_STACK_R_ADDR=0x68800000
 CONFIG_ARMV7_BOOT_SEC_DEFAULT=y
 CONFIG_SPL_TEXT_BASE=0x2F011000
+CONFIG_ANDROID_BOOT_IMAGE=y
```
@ldts Does AVB have some requirement for this?

mike-scott reviewed (commented):
Just the 1 config question on my end
